### ISSUE TITLE
Integration of css grid and flexbox?

### ISSUE THREAD

**Issue by funtoosh** [textpattern/textpattern-default-theme#85]
Phil, what do you think of integrating css grid and flexbox instead of float and clearfix? Is this something that you would use already, or do you prefer to wait for better browser support?

Cheers,
-martin

**Comment by philwareham** [comment]
I use flexbox for grids at the moment in some projects (http://hive-framework.philwareham.co.uk/#section-grid) but since the latest Textpattern default theme is single column, there is no need here I think.

(Regarding CSS grid for projects - it is too early to start implementing that IMO. Otherwise you'll have to write flexbox rules as a fallback anyway).

**Comment by funtoosh** [comment]
It's quite unfortunate that IE/Edge don't really support grid, but there is the IE legacy grid implementation: https://cm.engineering/getting-to-know-css-grid-layout-818e43ca71a5

Using flexbox as a fallback does not make sense at all though, you're right.

(Although the theme is single column, a lot of people will use it as a starting point, so having integrated a css grid would be a huge bonus, imho.)

**Comment by philwareham** [comment]
When @Bloke has implemented theme support in Textpattern I am planning on expanding the default theme into more of a mini-framework which will cover this issue (I'll also hopefully be providing themes based on other popular frameworks such as Bootstrap and Foundation for Sites too). So I'm not ruling it out entirely - just not now.

Yes we can eventually use CSS grid with IE fallbacks if IE11 stubbornly refuses to die. At the present moment Safari is a major stumbling block on using CSS grid instead of flexbox - since Apple don't follow the 6 weekly update cycle that Firefox and Chrome adhere to, preferring to release browser updates within system updates.

**Comment by funtoosh** [comment]
Oh, Safari was actually updated to 10.1 the other day, and now supports css grid : )

Theme support sounds great, looking forward to that!

![bildschirmfoto 158](https://cloud.githubusercontent.com/assets/506433/24447651/0255e878-1472-11e7-94c0-361f0b4cc163.png)

**Comment by philwareham** [comment]
Yep latest release of Safari indeed has CSS grid, but there are too many users on older Safari right now - which was my point. I give it a about a year before I think I can safely use CSS grid without having flexbox fallbacks.

**Comment by funtoosh** [comment]
Phil, any plans on incorporating your Hive framework's grid into the default theme? I think that would be a great addition to the versatility of the theme.

**Comment by philwareham** [comment]
Quite possibly - I want to expand the theme somewhat in Textpattern 4.8 now we have allowances for JavaScript and suchlike via custom assets. The grid would be very easy to include - it's just CSS - so I can look into it.

**Comment by funtoosh** [comment]
Yep, that would be great!!
